direct/core/src/test/java/cz/o2/proxima/direct/storage/InMemStorage.java

```diff
@@ -465,10 +486,12 @@ private void handleFlushDataBaseOnPosition(
               .sorted(Comparator.comparingLong(a -> a.getValue().getFirst()))
               .forEachOrdered(
                   e -> {
-                    if (restartedOffset.getAndIncrement() < offset.getOffset()) {
+                    String keyAttr = e.getKey().substring(prefixLength);
+                    long stamp = e.getValue().getFirst();
+                    if (offset.getConsumedKeyAttr().contains(keyAttr + ":" + stamp)) {
```

Comment:
can you please describe under what circumstances this can happen in the inline comment?